src/plan/aggregate.rs

```diff
@@ -102,68 +104,61 @@ impl<P: Implementable> Implementable for Aggregate<P> {
         // resulting collections, s.t. they can be joined afterwards.
         for (i, aggregation_fn) in self.aggregation_fns.iter().enumerate() {
             let value_offset = value_offsets[i];
+            let with_lenght = self.with_symbols.len();
+
+            // Access the right value for the given iteration loop and extend possible with-values.
+            let prepare_unary = move |(key, tuple): (Vec<Value>, Vec<Value>)| {
+                let value = &tuple[value_offset];
+                let mut v = vec![value.clone()];
+
+                // With-symbols are always the last elements in the value part of each tuple, given they are specified.
+                // We append these, s.t. we consolidate correctly.
+                // @TODO I'm not sure if the compiler is smart enough to skip the iteration if `with-lenght` is zero
+                v.extend(tuple.iter().rev().take(with_lenght).cloned());
+
+                (key, v)
+            };
 
             match aggregation_fn {
                 AggregationFn::MIN =>  {
                     let tuples = tuples
-                        .map(move |(key, tuple)| {
-                            let v = match tuple[value_offset] {
-                                Value::Number(num) => num,
-                                _ => panic!("MIN can only be applied on type Number."),
-                            };
-                            (key, v)
-                        })
+                        .map(prepare_unary)
                         .group(|_key, vals, output| {
-                            let min = vals[0].0;
-                            output.push((*min, 1));
+                            let min = &vals[0].0[0];
```

Comment:
Uh this might be a problem. The sort order that `group` maintains won't be the right one anymore with tuples, would it?

Reply:
It looks like it does the right thing. 
The order by which the values in `group` are arranged is descending left to right, the first value in each element being the primary key, the second the secondary and so on.
As we are only interested in the first value all is right.
Here is the example of the `with`-test but with `min` instead of `sum`
![image](https://user-images.githubusercontent.com/19425283/48556648-d9db1000-e8e4-11e8-892c-771c6d161139.png)

Reply:
The first 4 entries are the ones after `prepare_unary`
The last the `vals` in the `group` context

Reply:
![image](https://user-images.githubusercontent.com/19425283/48557170-4c98bb00-e8e6-11e8-9341-4a31166c0c2b.png)
This is the same example but with emtpy `with`

Reply:
Great :)